misk/src/main/kotlin/misk/hibernate/Query.kt

```diff
@@ -22,3 +22,20 @@ inline fun <reified P : Projection> Query<*>.uniqueResultAs(session: Session) =
     session, P::class)
 
 inline fun <reified T : Query<*>> Query.Factory.newQuery(): T = newQuery(T::class)
+
+/**
+ * Annotations a function on a subinterface of [Query] to indicate which column (or path of columns)
+ * it constrains and using which operator.
+ */
+annotation class Constraint(
+  val path: String,
+  val operator: String = "="
```

Comment:
should probably document the supported operators

Reply:
Yeah, Only supporting `=` and `<` is probably insufficient!